app/unit-tests/src/org/commcare/android/tests/formentry/IntentCalloutTests.java

```diff
@@ -0,0 +1,117 @@
+package org.commcare.android.tests.formentry;
+
+import android.content.Intent;
+import android.os.Environment;
+import android.widget.ImageButton;
+
+import junit.framework.Assert;
+
+import org.commcare.CommCareTestApplication;
+import org.commcare.activities.FormEntryActivity;
+import org.commcare.android.CommCareTestRunner;
+import org.commcare.android.resource.installers.XFormAndroidInstaller;
+import org.commcare.android.util.ActivityLaunchUtils;
+import org.commcare.android.util.TestAppInstaller;
+import org.commcare.dalvik.R;
+import org.commcare.views.widgets.IntegerWidget;
+import org.commcare.views.widgets.IntentWidget;
+import org.commcare.views.widgets.StringNumberWidget;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowActivity;
+import org.robolectric.shadows.ShadowEnvironment;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * @author wpride
+ */
+@Config(application = CommCareTestApplication.class)
+@RunWith(CommCareTestRunner.class)
+public class IntentCalloutTests {
+
+    @Before
+    public void setup() {
+        XFormAndroidInstaller.registerAndroidLevelFormParsers();
+        TestAppInstaller.installAppAndLogin(
+                "jr://resource/commcare-apps/phone_call/profile.ccpr",
+                "test", "123");
+        ShadowEnvironment.setExternalStorageState(Environment.MEDIA_MOUNTED);
+    }
+
+    /**
+     * Test different behaviors for possibly grouped intent callout views
+     */
+    @Test
+    public void testIntentCalloutWithData() {
```

Comment:
Can you rename this test to make it more obvious that this is the one testing the XPath format?